enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,437 @@
+  downgrades are not expected to present a significant challenge.
+
+### Version Skew Strategy
+
+TBD but I don't think it applies here
+
+### minimal Aspects of API Extensions
+
+TBD but I don't think it applies here
+
+#### Failure Modes
+
+TBD but I don't think it applies here
+
+#### Support Procedures
+
+TBD but I don't think it applies here
+
+## Implementation History
+
+Initial proofs-of-concept:
+
+- https://github.com/openshift/cluster-monitoring-operator/pull/1785
+
+## Alternatives
+
+- Make CMO injecting metric relabelling for all service monitors based on the
+  rules being deployed, but this is not a good idea because: 
+  - CMO acting behind the back of other operators isn't probably a very good
+    idea (less explicit, more brittle);
+  - it's likely to be less efficient because the relabeling configs would be
+    very complex and expensive in terms of processing.
+  - Hypershift implements a [similar
+    strategy](https://github.com/openshift/hypershift/pull/1294), but again this
+    would only work for service monitors that CMO deploys directly
+- Let users configure themselves the Prometheus scraping interval. This solution
+  was discarded because:
+  - Users might not be fully aware of the impact that changing this interval
+    might have. Too low of an interval and users might overwhelm Prometheus and
+    exhaust its memory. In contrast, too long of an interval might render the
+    default alerts ineffective;
+  - It's not advisable to increase the scrape interval to more than 2 minutes.
+    Staleness occurs at 5 minutes (by default) which is would cause gaps in
+    graphs/dashboards;
+  - Many upstream dashboards were built while having the 30 second interval in
+    mind;
+  - Scrapes can fail, but the user might not be mindful of this and set a high
+    scrape interval;
+- Add a seperate container to prometheus-operator (p-o) that would be used by
+  p-o to modify the prometheus config according to a metrics collection profile.
+  - This container would perform an analysis on what metrics were being used.
+    Then it would provide prometheus operator with this list.
+  - Prometheus-operator with the list provided by this new component would know
+    what scraping targets it could change to keep certain metrics.
+- Recently Azure also added support for metrics collection profiles:
+  - [Azure
+    Docs](https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-scrape-configuration-minimal)
+  -  https://github.com/Azure/prometheus-collector
+  - In their approach they also have [hardcoded](https://github.com/Azure/prometheus-collector/blob/66ed1a5a27781d7e7e3bb1771b11f1da25ffa79c/otelcollector/configmapparser/tomlparser-default-targets-metrics-keep-list.rb#L28)
+  set of metrics that are only consumed when the minimal profile is enabled.
+  However, customer are also able to extend this minimal profile with regexes to
+  include metrics which might be interesting to them.
+
```

Comment:
Add reference to https://docs.google.com/document/d/1I-YT7LKKDHSBLB6Hmg0tZ54DWjrAxlVdXxlViShMu-0/edit